contracts/utils/math/Math.sol

```diff
@@ -275,6 +283,24 @@ library Math {
         }
     }
 
+    /**
+     * @dev Returns the modular exponentiation of the specified base, exponent and modulus (b ** e % m)
+     *
+     * Requirements:
+     * - modulus can't be zero
+     * - result should be obtained successfully
+     */
+    function modExp(uint256 b, uint256 e, uint256 m) internal view returns (uint256) {
+        if (m == 0) {
+            revert MathModulusEqualsZero();
+        }
+        (bool success, bytes memory result) = (address(5).staticcall(abi.encode(32, 32, 32, b, e, m)));
+        if (!success) {
+            revert MathModExpCannotBeCalculated();
+        }
```

Comment:
Couldn't find a way to make the precompile fail, not even with `m == 0`. I looked at:

- [EIP 198](https://eips.ethereum.org/EIPS/eip-198)
- [EVM Codes](https://www.evm.codes/precompiled?fork=shanghai)

I'll try sending less gas than required to test this error.

Reply:
Didn't work, I tried different gasLimit values in the following test;

```
 it.only('reverts with not enough gas', async function () {
  await expect(
    this.mock.$modExp(
      0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffn,
      0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffn,
      0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffn,
      {
        gasLimit: 25_000,
      },
    ),
  ).to.be.revertedWithCustomError(this.mock, 'MathModExpCannotBeCalculated');
});
```

It reverts as out of gas but doesn't trigger the `MathModExpCannotBeCalculated`.

Reply:
yes, I guess that is because if the precompile has not enough gas, than the 1/64 that remains after that are not enough to check success and revert with a custom error